### PR TITLE
Add support for Overlayfs.

### LINKED ISSUE #932: Support overlayfs
#926 adds support for docker image storage partition discovery with a device mapper storage backend. We need to add support for Overlayfs. This is a requirement for Kubernetes v1.1 to work on OS distros that use Overlayfs as the default storage backend driver.

cc @yifan-gu cc @dchen1107

### PR DESCRIPTION
**Pull request by vishh**
Fixes #932

### CODE REVIEW

**Comment by jimmidyson** [comment]
retest this please

**Comment by jimmidyson** [comment]
Do you think it's time to put FS stats behind an interface & create appropriate implementation on startup? Would remove some of the switches & perhaps make it more extensible & easier to test going forward?

**Comment by jimmidyson** [comment]
@vishh Any time to put FS stats behind an interface with separate impls?

**Comment by jimmidyson** [comment]
@vishh Also needs an integration test set up for it please.

Is this PR still alive btw?

**Comment by vishh** [comment]
@jimmidyson: An interface sounds good. I'd prefer performing the refactoring in a separate PR though.

**Comment by jimmidyson** [comment]
@vishh OK so can you rebase & let's try to sort out integration tests ready to merge this.

**Comment by vishh** [comment]
Rebased to HEAD.

Since devicemapper support is pending, integration tests will need to
understand the underlying docker config.

Are you planning on adding support for devicemapper anytime soon?

**Comment by jimmidyson** [comment]
It won't be this week sadly.

**Comment by jimmidyson** [comment]
Can you rebase plesae?

**Comment by vishh** [comment]
Rebased.

**Comment by jimmidyson** [comment]
LGTM. Thanks @vishh.

File: container/docker/factory.go [hunk 1]
```diff
@@ -23,13 +23,13 @@ import (
...
 	"sync"
 
+	"github.com/fsouza/go-dockerclient"
 	"github.com/google/cadvisor/container"
 	"github.com/google/cadvisor/container/libcontainer"
 	"github.com/google/cadvisor/fs"
 	info "github.com/google/cadvisor/info/v1"
 	"github.com/google/cadvisor/utils"
 
-	"github.com/fsouza/go-dockerclient"
```
Comment (carmark) [inline container/docker/factory.go]: It looks like a merge issue, shouldn't change it.

Comment (vishh) [inline container/docker/factory.go]: How is this ordering guaranteed? If its through a special golang tool, that should be part of the build infra.

Comment (vishh) [inline container/docker/factory.go]: In any case, I fixed it for now.

**Comment by vishh** [comment]
Thanks for the review. Self merging now..

**Pull request merged**